scripts/test_env.py

```diff
@@ -0,0 +1,191 @@
+            shutil.copy2("./configs/peer/genesis.json", peers_dir)
+            shutil.copy2("./configs/peer/validator.wasm", peers_dir)
+        except FileNotFoundError:
+            logging.error("Some of the config files are missing. \
+                          Please provide them in the `./configs/peer` directory")
+            sys.exit(1)
+        try:
+            shutil.copy2("./target/debug/iroha", peers_dir)
+        except FileNotFoundError:
+            logging.error("Please build the `iroha` by running `cargo build --bin iroha`")
+            sys.exit(1)
+
+        self.peers = [Peer(setup_dir, host_addr, i) for i in range(n_peers)]
+
+        os.environ["IROHA2_CONFIG_PATH"] = str(setup_dir.joinpath("peers/config.json"))
+        os.environ["IROHA2_GENESIS_PATH"] = str(setup_dir.joinpath("peers/genesis.json"))
+        os.environ["IROHA_GENESIS_ACCOUNT_PUBLIC_KEY"] = self.peers[0].public_key
+        os.environ["IROHA_GENESIS_ACCOUNT_PRIVATE_KEY"] = self.peers[0].private_key
+
+        self.generate_trusted_peers()
+    
+    def generate_trusted_peers(self):
+        self.trusted_peers = []
+        for peer in self.peers:
+            self.trusted_peers.append(f'{{"address": "{peer.host_addr}:{peer.p2p_port}", "public_key": "{peer.public_key}"}}')
+        os.environ["SUMERAGI_TRUSTED_PEERS"] = f"[{','.join(self.trusted_peers)}]"
+
+    def run(self):
+        self.peers[0].run(is_genesis=True)
+        for peer in self.peers[1:]:
+            peer.run()
+
+class Peer:
+    """
+    A single bootstrapped peer. Could be a genesis node or a regular peer.
+    Should not be run directly, but rather as a part of a Network.
+    """
+    def __init__(self, setup_dir: pathlib.Path, host_addr: ipaddress.IPv4Address, nth: int):
+        self.nth = nth
+        self.name = f"iroha{nth}"
+        self.p2p_port = 1337 + nth
+        self.api_port = 8080 + nth
+        self.telemetry_port = 8180 + nth
+        self.tokio_console_port = 5555 + nth
+        self.setup_dir = setup_dir
+        self.host_addr = host_addr
+        self.generate_key_pair()
+        logging.info("Peer %s generated key pair", self.name)
+        
+    def generate_key_pair(self):
+        kagami = subprocess.run([f"{self.setup_dir}/kagami","crypto", "-j"], capture_output=True)
+        if kagami.returncode:
+            logging.error("Kagami failed to generate a key pair.")
+            sys.exit(1)
+        str_keypair = kagami.stdout
+        json_keypair = json.loads(str_keypair)
+        # public key is a string, private key is a json object
+        self.public_key = json_keypair['public_key']
+        self.private_key = json.dumps(json_keypair['private_key'])
+
+    def run(self, is_genesis: bool = False):
+        logging.info("Running peer %s...", self.name)
+        peer_dir = self.setup_dir.joinpath(f"peers/{self.name}")
+        os.mkdir(peer_dir)
+        os.mkdir(peer_dir.joinpath("storage"))
```

Comment:
ditto about path joins in this function

Reply:
Ditto about Python being weird and often problematic.